service/matching/matching_engine.go

```diff
@@ -1296,26 +1299,34 @@ func (e *matchingEngineImpl) DescribeTaskQueue(
 		if !pm.Partition().IsRoot() {
 			return nil, serviceerror.NewInvalidArgument("DescribeTaskQueue stats are only supported for the root partition")
 		}
-		cacheKey := ".taskQueueStats" // starts with `.` to prevent clashing with build ID cache keys
+
+		var buildIds []string
+		if request.Version != nil {
+			// A particular version was requested. This is only available internally; not user-facing.
+			buildIds = []string{worker_versioning.WorkerDeploymentVersionToStringV31(request.Version)}
+		}
+
+		cacheKey := "dtq_default:" + strings.Join(buildIds, ",")
```

Comment:
This needs to be keyed by the requested version or the cache results of all vs one version conflict.

Reply:
Not for this PR, but could we always put the per-version stats in the cache, keyed by version string so they do not collide? It'd mean the total result here would need to check cache for all individual versions but that should be fine, right?

Reply:
👍 Good idea.